src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -20,10 +20,18 @@ class GFAL2Impl(StageOutImpl):
 
     def __init__(self, stagein=False):
         StageOutImpl.__init__(self, stagein)
-        self.setups = "env -i X509_USER_PROXY=$X509_USER_PROXY; "
-        self.setups += 'date "+%Y-%m-%dT%H:%M:%S";'
-        self.removeCommand = "%s gfal-rm -vvv -t 600 " % self.setups
-        self.copyCommand = "%s gfal-copy -vvv -t 2400 -T 2400 -p " % self.setups
+        # %s in the end is used for adding removeCommand or copyCommand. If we want to execute
+        # commands in clean shell, we can`t separate them with ';'. Next commands after separation
+        # are executed without env -i and this leads us with mixed environment with COMP and system python.
+        # GFAL2 is not build under COMP environment and it had failures with mixed environment.
+        self.setups = "env -i X509_USER_PROXY=$X509_USER_PROXY sh -c '%s'"
+        # %s in the end is used for adding file location which is needed to be removed
+        self.removeCommand = self.setups % 'printenv; date; gfal-rm -vvv -t 600 %s '
+        # %(checksum) is adding a checksum option
+        # %(option) is adding options which are defined in local site conf
+        # %(source) is source file location
+        # %(destination) is destination file location
+        self.copyCommand = self.setups % 'printenv; date; gfal-copy -vvv -t 2400 -T 2400 -p %(checksum)s %(options)s %(source)s %(destination)s'
```

Comment:
I think this code is over documented. (Probably the first time I've said this). If you are doing something tricky, by all means document it, but I don't think you need to specify all the parameters since that's pretty obvious.